### PR TITLE
Make code more compatible with older versions of Hive

### PR DESCRIPTION
**Pull request by jphalip**

### CODE REVIEW

File: connector/src/main/java/com/google/cloud/hive/bigquery/connector/input/avro/AvroSerializer.java [hunk 1]
```diff
@@ -145,7 +144,7 @@ public static Object serialize(
 
     if (objectInspector instanceof HiveDecimalObjectInspector) {
       byte[] bytes = ((ByteBuffer) avroObject).array();
-      int scale = (int) actualSchema.getObjectProp("scale");
+      int scale = actualSchema.getJsonProp("scale").asInt();
```
Comment (yigress) [inline connector/src/main/java/com/google/cloud/hive/bigquery/connector/input/avro/AvroSerializer.java:147]: this getJsonProp won't work with newer avro versions? in avro-1.11 it is a private method

Comment (jphalip) [reply connector/src/main/java/com/google/cloud/hive/bigquery/connector/input/avro/AvroSerializer.java]: I've added a workaround for this in the latest commit. Please take a look

**Comment by yigress** [comment]
[INFO] Scanning for projects...
Downloading from central: https://repo.maven.apache.org/maven2/com/google/cloud/google-cloud-storage-bom/2.20.1/google-cloud-storage-bom-2.20.1.pom
Progress (1): 2.7/3.6 kB
Downloading from central: https://repo.maven.apache.org/maven2/com/google/cloud/google-cloud-shared-config/1.5.5/google-cloud-shared-config-1.5.5.pom
Progress (1): 2.7/30 kB
Downloading from central: https://repo.maven.apache.org/maven2/com/google/protobuf/protobuf-bom/3.21.12/protobuf-bom-3.21.12.pom
Progress (1): 2.7/3.5 kB
Downloading from central: https://repo.maven.apache.org/maven2/io/grpc/grpc-bom/1.53.0/grpc-bom-1.53.0.pom
Progress (1): 2.7/5.4 kB
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-common:jar must be a valid version but is '${hadoop.version}'. @ com.google.cloud.hive:shaded-dependencies:${revision}, /workspace/shaded-dependencies/pom.xml, line 54, column 16
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-mapreduce-client-common:jar must be a valid version but is '${hadoop.version}'. @ com.google.cloud.hive:shaded-dependencies:${revision}, /workspace/shaded-dependencies/pom.xml, line 66, column 16
[ERROR] 'dependencies.dependency.version' for org.apache.hive:hive-exec:jar must be a valid version but is '${hive.version}'. @ com.google.cloud.hive:shaded-dependencies:${revision}, /workspace/shaded-dependencies/pom.xml, line 82, column 16
[ERROR] 'dependencies.dependency.version' for org.apache.hive:hive-common:jar must be a valid version but is '${hive.version}'. @ com.google.cloud.hive:shaded-dependencies:${revision}, /workspace/shaded-dependencies/pom.xml, line 94, column 16
[ERROR] 'dependencies.dependency.version' for org.apache.hive:hive-service:jar must be a valid version but is '${hive.version}'. @ com.google.cloud.hive:shaded-dependencies:${revision}, /workspace/shaded-dependencies/pom.xml, line 100, column 16
[ERROR] 'dependencies.dependency.version' for org.apache.tez:tez-dag:jar must be a valid version but is '${tez.version}'. @ com.google.cloud.hive:shaded-dependencies:${revision}, /workspace/shaded-dependencies/pom.xml, line 113, column 16
[ERROR] 'dependencies.dependency.version' for org.apache.tez:tez-common:jar must be a valid version but is '${tez.version}'. @ com.google.cloud.hive:shaded-dependencies:${revision}, /workspace/shaded-dependencies/pom.xml, line 125, column 16
[ERROR] 'dependencies.dependency.version' for org.apache.tez:tez-mapreduce:jar must be a valid version but is '${tez.version}'. @ com.google.cloud.hive:shaded-dependencies:${revision}, /workspace/shaded-dependencies/pom.xml, line 137, column 16
[ERROR] 'dependencies.dependency.version' for com.google.cloud.bigdataoss:gcs-connector:jar:shaded must be a valid version but is '${gcs-connector.version}'. @ com.google.cloud.hive:hive-bigquery-connector:${revision}, /workspace/connector/pom.xml, line 82, column 16
 @ 
[ERROR] The build could not read 2 projects -> [Help 1]
[ERROR]   
[ERROR]   The project com.google.cloud.hive:shaded-dependencies:2.0.0-SNAPSHOT (/workspace/shaded-dependencies/pom.xml) has 8 errors
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-common:jar must be a valid version but is '${hadoop.version}'. @ com.google.cloud.hive:shaded-dependencies:${revision}, /workspace/shaded-dependencies/pom.xml, line 54, column 16
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-mapreduce-client-common:jar must be a valid version but is '${hadoop.version}'. @ com.google.cloud.hive:shaded-dependencies:${revision}, /workspace/shaded-dependencies/pom.xml, line 66, column 16
[ERROR]     'dependencies.dependency.version' for org.apache.hive:hive-exec:jar must be a valid version but is '${hive.version}'. @ com.google.cloud.hive:shaded-dependencies:${revision}, /workspace/shaded-dependencies/pom.xml, line 82, column 16
[ERROR]     'dependencies.dependency.version' for org.apache.hive:hive-common:jar must be a valid version but is '${hive.version}'. @ com.google.cloud.hive:shaded-dependencies:${revision}, /workspace/shaded-dependencies/pom.xml, line 94, column 16
[ERROR]     'dependencies.dependency.version' for org.apache.hive:hive-service:jar must be a valid version but is '${hive.version}'. @ com.google.cloud.hive:shaded-dependencies:${revision}, /workspace/shaded-dependencies/pom.xml, line 100, column 16
[ERROR]     'dependencies.dependency.version' for org.apache.tez:tez-dag:jar must be a valid version but is '${tez.version}'. @ com.google.cloud.hive:shaded-dependencies:${revision}, /workspace/shaded-dependencies/pom.xml, line 113, column 16
[ERROR]     'dependencies.dependency.version' for org.apache.tez:tez-common:jar must be a valid version but is '${tez.version}'. @ com.google.cloud.hive:shaded-dependencies:${revision}, /workspace/shaded-dependencies/pom.xml, line 125, column 16
[ERROR]     'dependencies.dependency.version' for org.apache.tez:tez-mapreduce:jar must be a valid version but is '${tez.version}'. @ com.google.cloud.hive:shaded-dependencies:${revision}, /workspace/shaded-dependencies/pom.xml, line 137, column 16
[ERROR]   
[ERROR]   The project com.google.cloud.hive:hive-bigquery-connector:2.0.0-SNAPSHOT (/workspace/connector/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for com.google.cloud.bigdataoss:gcs-connector:jar:shaded must be a valid version but is '${gcs-connector.version}'. @ com.google.cloud.hive:hive-bigquery-connector:${revision}, /workspace/connector/pom.xml, line 82, column 16
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException

**Comment by jphalip** [comment]
@yigress The checks are now passing

**Review by yigress — APPROVED** [review]

**Pull request merged**